### PR TITLE
Improve debug logs

### PR DESCRIPTION
**Pull request by FelipeRosa**

### CODE REVIEW

File: src/connection.rs [hunk 1]
```diff
@@ -103,18 +103,28 @@ impl Connection {
...
-                                    assert!(matches!(
-                                        self.pending_requests.insert(msg.reference, sender),
-                                        None
-                                    ));
+                                    if self
+                                        .pending_requests
+                                        .insert(msg.reference, sender)
+                                        .is_some()
+                                    {
+                                        log::debug!("Replaced request with same ref");
```
Comment (lsunsi) [inline src/connection.rs]: Maybe warn? Just thinking, can this happen in a correct-usage of the lib? Also, maybe add the ref in the log?

File: src/connection.rs [hunk 2]
```diff
@@ -103,18 +103,28 @@ impl Connection {
...
+                                    }
                                 }
                                 Err(e) => {
-                                    let _ = sender.send(Err(RequestSendError::Send(e)));
+                                    if sender.send(Err(RequestSendError::Send(e))).is_err() {
+                                        log::debug!("Internal request channel receiver dropped");
```
Comment (lsunsi) [inline src/connection.rs:116]: When can this happen?

File: src/connection.rs [hunk 3]
```diff
@@ -103,18 +103,28 @@ impl Connection {
...
                                 }
                             }
                         }
                         Err(e) => {
-                            let _ = sender.send(Err(RequestSendError::InvalidMessageOutJson(e)));
+                            if sender
+                                .send(Err(RequestSendError::InvalidMessageOutJson(e)))
+                                .is_err()
+                            {
+                                log::debug!("Internal request channel receiver dropped");
```
Comment (lsunsi) [inline src/connection.rs:126]: When can this happen?

File: src/connection.rs [hunk 4]
```diff
@@ -164,9 +174,13 @@ impl Connection {
                                         Some(reference) => {
                                             match self.pending_requests.remove(reference) {
                                                 Some(sender) => {
-                                                    let _ = sender.send(Ok(msg));
+                                                    if sender.send(Ok(msg)).is_err() {
+                                                        log::debug!("Internal request channel receiver dropped");
+                                                    }
+                                                }
+                                                None => {
+                                                    log::debug!("Received response for a request that was not pending");
```
Comment (lsunsi) [inline src/connection.rs:182]: Maybe add the ref?

File: src/connection.rs [hunk 5]
```diff
@@ -164,9 +174,13 @@ impl Connection {
                                         Some(reference) => {
                                             match self.pending_requests.remove(reference) {
                                                 Some(sender) => {
-                                                    let _ = sender.send(Ok(msg));
+                                                    if sender.send(Ok(msg)).is_err() {
+                                                        log::debug!("Internal request channel receiver dropped");
```
Comment (lsunsi) [inline src/connection.rs:178]: When can this happen?

File: src/connection.rs (hunk 1 above)
Comment (FelipeRosa) [reply src/connection.rs]: Would only happen if there was a bug in the lib

File: src/connection.rs (hunk 2 above)
Comment (FelipeRosa) [reply src/connection.rs:116]: Edit: If the connection closed while making a request

File: src/connection.rs (hunk 3 above)
Comment (FelipeRosa) [reply src/connection.rs:126]: If the request payload serialization failed

File: src/connection.rs (hunk 5 above)
Comment (FelipeRosa) [reply src/connection.rs:178]: If the client dropped the response receiver while making a request

File: src/connection.rs (hunk 1 above)
Comment (lsunsi) [reply src/connection.rs]: Aaaaa, gg. Maybe error then?

**Review by lsunsi — APPROVED** [review]

**Pull request merged**